### PR TITLE
Add width and height attributes to cover images

### PR DESCRIPTION
**Pull request by igruenig**
**What does this PR change? What problem does it solve?**

When using cover images the layout shifts on initial load because the image dimensions can only be determined by the browser **after** loading the images.

I followed the recommendations from this [website](https://web.dev/optimize-cls/?utm_source=lighthouse&utm_medium=devtools#images-without-dimensions) linked to from Lighthouse in Chrome and added the width and height attributes to the cover images. From this the browser can determine the aspect ratio **before** loading any images.

Additionally, in the CSS file I have added the `height: auto` directive to the `.entry-cover img` selector. The height is then calculated based on the actual available width.

**Was the change discussed in an issue or in the Discussions before?**

Fix for #499 in the Discussions

## PR Checklist

- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] This change **does not** include any CDN resources/links.
- [x] This change **does not** include any unrelated scripts such as bash and python scripts.

### CODE REVIEW

File: layouts/partials/cover.html [hunk 1]
```diff
@@ -11,19 +11,32 @@
         {{- $processableFormats := (slice "jpg" "jpeg" "png" "tif" "bmp" "gif") }}
         {{- $prod := (hugo.IsProduction | or (eq .Site.Params.env "production")) }}
         {{- if (and (in $processableFormats $cover.MediaType.SubType) (ne .Site.Params.cover.responsiveImages false) (eq $prod true)) }}
-        <img loading="lazy" srcset="{{- range $size := $sizes -}}
-                        {{- if (ge $cover.Width $size) -}}
+        <img 
+            loading="lazy" 
+            srcset="{{- range $size := $sizes -}}
+                    {{- if (ge $cover.Width $size) -}}
```
Comment (danielfdickinson) [inline layouts/partials/cover.html:17]: This is an 'unrelated change'. It's not necessary and means that no upsizing occurs, where it did in the past. I know upsizing doesn't produce ideal images, but this should at least be a change that one opts into rather than having unexpectedly occur.

File: layouts/partials/cover.html [hunk 2]
```diff
@@ -11,19 +11,32 @@
...
-                    {{- end -}}{{$cover.Permalink }} {{printf "%dw" ($cover.Width)}}"
-            sizes="(min-width: 768px) 720px, 100vw" src="{{ $cover.Permalink }}" alt="{{ $alt }}" />
+                    {{ end }}
+                {{- end -}}{{$cover.Permalink }} {{printf "%dw" ($cover.Width)}}"
+            sizes="(min-width: 768px) 720px, 100vw"
+            src="{{ $cover.Permalink }}"
+            alt="{{ $alt }}"
+            width="{{ $cover.Width }}"
+            height="{{ $cover.Height }}"
+        />
```
Comment (danielfdickinson) [inline layouts/partials/cover.html:26]: This is a minor nit, but if img tags elsewhere in source do not having the (optional) '/' then it shouldn't be added here either. I know some linters insist on it, but others insist on removing. The 'right' answer depends on the existing code base.

File: layouts/partials/cover.html [hunk 3]
```diff
@@ -11,19 +11,32 @@
...
         {{- end }}
     {{- else }}{{/* For absolute urls and external links, no img processing here */}}
         {{- if $addLink }}<a href="{{ (.Params.cover.image) | absURL }}" target="_blank"
             rel="noopener noreferrer">{{ end -}}
-            <img loading="lazy" src="{{ (.Params.cover.image) | absURL }}" alt="{{ $alt }}">
+            <img loading="lazy" src="{{ (.Params.cover.image) | absURL }}" alt="{{ $alt }}"/>
```
Comment (danielfdickinson) [inline layouts/partials/cover.html:39]: Likewise here.

**Review by danielfdickinson — CHANGES_REQUESTED** [review]
This are of course my opinions and may not be gospel truth.

File: layouts/partials/cover.html (hunk 3 above)
Comment (igruenig) [reply layouts/partials/cover.html:39]: Thank you, missed those.

Comment (igruenig) [reply layouts/partials/cover.html:39]: I have no strong opinion on closing tags, I will remove them.

File: layouts/partials/cover.html (hunk 1 above)
Comment (igruenig) [reply layouts/partials/cover.html:17]: Maybe I miss the point here, but I only reformatted that part. The size check was there before my change.

Comment (danielfdickinson) [reply layouts/partials/cover.html:17]: Oh, you are right, I missed that. I think the reformatting should be in a separate commit or PR, unless absolutely necessary

Comment (igruenig) [reply layouts/partials/cover.html]: Agreed. I reduced the changes to a minimum.

Comment (danielfdickinson) [reply layouts/partials/cover.html]: Thank you. Looks good to me now. Now to see what @adityatelange thinks...

Comment (igruenig) [reply layouts/partials/cover.html]: Thanks for reviewing!

**Review by danielfdickinson — COMMENTED** [review]
Thank you for the changes. I'm not sure what I think of the formatting changes, mostly because I'm not sure what the maintainer prefers, but this looks good. Also, there is an issue #499 you might want to reference.

**Comment by igruenig** [comment]
Thank you for the hint! I inserted the link in my initial comment.

File: layouts/partials/cover.html [hunk 4]
```diff
@@ -15,10 +15,12 @@
...
                         {{ end }}
-                    {{- end -}}{{$cover.Permalink }} {{printf "%dw" ($cover.Width)}}"
-            sizes="(min-width: 768px) 720px, 100vw" src="{{ $cover.Permalink }}" alt="{{ $alt }}" />
+                    {{- end -}}{{$cover.Permalink }} {{printf "%dw" ($cover.Width)}}" 
+            sizes="(min-width: 768px) 720px, 100vw" src="{{ $cover.Permalink }}" alt="{{ $alt }}" 
+            width="{{ $cover.Width }}" height="{{ $cover.Height }}">
         {{- else }}{{/* Unprocessable image or responsive images disabled */}}
-        <img loading="lazy" src="{{ (path.Join .RelPermalink .Params.cover.image) | absURL }}" alt="{{ $alt }}">
+        <img loading="lazy" src="{{ (path.Join .RelPermalink .Params.cover.image) | absURL }}" alt="{{ $alt }}" 
+            width="{{ $cover.Width }}" height="{{ $cover.Height }}">
```
Comment (adityatelange) [inline layouts/partials/cover.html:23]: `Width` and `Height` won't exist for `Unprocessable image` right?

Comment (igruenig) [reply layouts/partials/cover.html]: Good point. I removed width and height for unprocessable images.

**Review by adityatelange — APPROVED** [review]
